### PR TITLE
Additional cfg files

### PR DESCRIPTION
**Pull request by dankwart-de**
First version, needs discussion.
Tests for BashConvert toXML work.

### CODE REVIEW

File: dist/plugins/DefaultPlugin/DefaultPlugin.iml [hunk 1]
```diff
@@ -14,9 +14,9 @@
     </content>
     <orderEntry type="inheritedJdk" />
     <orderEntry type="sourceFolder" forTests="false" />
-    <orderEntry type="module" module-name="roddy" />
     <orderEntry type="module" module-name="PluginBase" />
     <orderEntry type="library" name="roddyBundledLibraries" level="project" />
+    <orderEntry type="library" name="roddy" level="project" />
```
Comment (vinjana) [inline dist/plugins/DefaultPlugin/DefaultPlugin.iml]: No. Also roddy is a module.

File: RoddyCore/src/de/dkfz/roddy/config/ConfigurationFactory.groovy [hunk 2]
```diff
@@ -192,6 +194,27 @@ public class ConfigurationFactory {
...
+            return loadAndPreprocessBashFile(file.text)
+
+        if (file.name.endsWith(".yml")) // YAML import
+            return loadAndPreprocessYAMLFile(file.text)
+
+        throw new IOException("Unknown file type ${file.name}")
+    }
+
+    static String loadAndPreprocessYAMLFile(String s) {
+        return new YAMLConverter().convertToXML(s)
```
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/config/ConfigurationFactory.groovy]: The yaml part is far from finished. Now it is only about the Bash configs.

File: RoddyCore/src/de/dkfz/roddy/config/ConfigurationFactory.groovy [hunk 3]
```diff
@@ -192,6 +194,27 @@ public class ConfigurationFactory {
...
+        if (file.name.endsWith(".xml")) // Default behaviour
+            return file.text
+
+        if (file.name.endsWith(".sh")) // Easy Bash importer
+            return loadAndPreprocessBashFile(file.text)
+
+        if (file.name.endsWith(".yml")) // YAML import
+            return loadAndPreprocessYAMLFile(file.text)
+
+        throw new IOException("Unknown file type ${file.name}")
```
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/config/ConfigurationFactory.groovy]: I think it is a good idea to start creating our own exceptions. I'll create one for this case.

File: RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy [hunk 4]
```diff
@@ -335,4 +336,106 @@ class BashConverter extends ConfigurationConverter {
...
+
+    /**
+     * The easy Bash config format.
+     *
+     * This is a very generic helper script which just converts a bash cfg to xml.
+     * Not everything is covered, it's not built in yet and it needs a lot of improvement.
+     * But for now, it works.
+     * To run it faster from within IDEA, copy the content to a groovy console and start it.
+     *
+     * Basically a file which contains some info in the header and only config values.
```
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy]: I put in the example from the Test class.

File: RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy [hunk 5]
```diff
@@ -335,4 +336,106 @@ class BashConverter extends ConfigurationConverter {
...
+        xmlLines << "description='" + getHeaderValue(header, "description", "") + "'"
+        xmlLines << "usedresourcessize='" + getHeaderValue(header, "usedresourcessize", "l") + "' >"
+
+        xmlLines << "  <availableAnalyses>"
+        getHeaderValues(header, "analysis", []).each {
+            String analysis ->
+                String[] split = analysis.split(StringConstants.COMMA)
+                String id = split[0]
+                String cfg = split[1]
+                String plg = split[2]
```
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy]: I added a check and further cleaned up the conversion method. As the file is read by the xml loader in a later step, the analysis will be ignored (and a logger message will be added).

File: RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy [hunk 6]
```diff
@@ -335,4 +336,106 @@ class BashConverter extends ConfigurationConverter {
...
+        // Convert the configuration values.
+        xmlLines << "  <configurationvalues>"
+
+        allLines[header.size()..-1].each {
+            String[] s = it.split("[=]")
+            if (!it) return;
+            if (!s) return;
+
+            if (isBashPipe(it)) {
+
```
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy]: nop? What do you mean?

File: RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy [hunk 7]
```diff
@@ -335,4 +336,106 @@ class BashConverter extends ConfigurationConverter {
...
+                    xmlLines << ("""<!-- ${it}""").toString()
+                }
+            } else {
+
+                if (isBashComment(previousLine)) {
+                    xmlLines[-1] += ("""-->""").toString()
+                }
+
+                if (s.size() == 2)
+                    xmlLines << ("""    <cvalue name='${s[0]}' value='${s[1]}' type='string' />""").toString()
```
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy]: hmm... no. But we can add a task for this and create the code with tests later on.

File: RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy [hunk 8]
```diff
@@ -335,4 +336,106 @@ class BashConverter extends ConfigurationConverter {
...
+        if (!found) return defaultValue
+        return found.collect { String entry -> entry.split(StringConstants.WHITESPACE)[1] }
+    }
+
+    public static boolean isBashComment(String it) {
+        it.trim().startsWith("#")
+    }
+
+    public static boolean isBashPipe(String it) {
+        it.trim().startsWith("###")
```
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy]: That is a special form of code only used for one of our internal tools. I'll show you later. Normally it is not of use.

File: RoddyCore/src/de/dkfz/roddy/config/converters/YAMLConverter.groovy [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    @Override
+    StringBuilder convertConfigurationValue(ConfigurationValue cv, ExecutionContext context) {
+        throw new NotImplementedException();
+    }
+
+    @Override
+    String convertToXML(String text) {
+        return null
```
Comment (dankwart-de) [inline RoddyCore/src/de/dkfz/roddy/config/converters/YAMLConverter.groovy]: It is not finished. I don't know how to do it yet.

File: dist/plugins/DefaultPlugin/DefaultPlugin.iml (hunk 1 above)
Comment (dankwart-de) [inline dist/plugins/DefaultPlugin/DefaultPlugin.iml]: Should be fixed.

File: RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy (hunk 6 above)
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/config/converters/BashConverter.groovy]: "No Operation"?

**Pull request merged**